pkg/endpoint/endpoint.go

```diff
@@ -904,8 +904,9 @@ func parseEndpoint(ctx context.Context, owner regeneration.Owner, policyGetter p
 
 	ep.initDNSHistoryTrigger()
 
-	// Validate the options that were parsed
-	ep.SetDefaultOpts(ep.Options)
+	// Set default options, unsupported options were already dropped by
+	// ep.Options.UnmarshalJSON
```

Comment:
think this should be `ep.UnmarshalJSON`,  ep.Options doesn't appear to have a UnmarshalJSON func.

Reply:
It's here:  https://github.com/cilium/cilium/pull/33164/files#diff-87bc5ccc7580502f71b247772a6e53aec7835f895cfaf90735895630fae75b5eR177-R195